Switch back to root to call quotactl(2), our kernel has too strict
permision checks.

Index: source3/lib/sysquotas_4B.c
--- source3/lib/sysquotas_4B.c.orig
+++ source3/lib/sysquotas_4B.c
@@ -51,7 +51,7 @@
 #include <jfs/quota.h>
 #endif
 
-#if defined(DARWINOS)
+#if defined(DARWINOS) || defined(__OpenBSD__)
 /* WorkARound broken HFS access checks in hfs_quotactl. Darwin only(?) */
 #define HFS_QUOTACTL_WAR 1
 #endif
